pycaret/internal/pycaret_experiment/time_series_experiment.py

```diff
@@ -851,6 +851,241 @@ def check_fh(self, fh: Union[List[int], int, np.array]) -> np.array:
             )
         return fh
 
+    @staticmethod
+    def __check_and_clean_data(data: Union[pd.Series, pd.DataFrame]) -> pd.DataFrame:
+        """Check that the data is of the correct type (Pandas Series or DataFrame).
+        Also cleans the data before coercing it into a dataframe which is used
+        internally for all future tasks.
+
+        Parameters
+        ----------
+        data : Union[pd.Series, pd.DataFrame]
+            Input data
+
+        Returns
+        -------
+        pd.DataFrame
+            Checked and Cleaned version of the data
+
+        Raises
+        ------
+        ValueError
+            Raised if data is not of the correct type
+
+        """
+        if not isinstance(data, (pd.Series, pd.DataFrame)):
+            raise ValueError(
+                f"Data must be a pandas Series or DataFrame, got object of {type(data)} type!"
+            )
+
+        ## Make a local copy (to perfrom inplace operation on the original dataset)
+        data_ = data.copy()
+
+        if isinstance(data_, pd.DataFrame):
+            if data_.shape[1] != 1:
+                raise ValueError(
+                    f"Currently only Univariate forecasting without exogenous variables is supported. ",
+                    f"Hence if data is a pandas DataFrame, it must have only 1 column. Got {data_.shape[1]} columns!",
+                )
```

Comment:
```suggestion
        if isinstance(data_, pd.DataFrame):
            if data_.shape[1] != 1:
                raise ValueError(
                    f"Currently only Univariate forecasting without exogenous variables is supported. ",
                    f"Hence if data is a pandas DataFrame, it must have only 1 column. Got {data_.shape[1]} columns!",
                )
            ## Make a local copy (to perfrom inplace operation on the original dataset)
            data_ = data.copy()
```

Reply:
no need for copy if a df is created from series

Reply:
Well, we are changing the series name so want to avoid the inplace operation.